### PR TITLE
Gradle Updates

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Upgrade to AGP 8 (and associated required project changes)
 - Update CI to use Java 17 (required for AGP 8)
 - Bump Kotlin version to 1.9.10
 - Update all dependencies to latest version
 - Cleanup unused dependencies

 ### Checklist

 - [x] Added a changelog entry

- @sarahkoop

### CODE REVIEW

File: AmericanExpress/build.gradle [hunk 1]
```diff
@@ -5,7 +5,8 @@ plugins {
 }
 
 android {
-    compileSdkVersion rootProject.compileSdkVersion
+    namespace "com.braintreepayments.api.americanexpress"
```
Comment (sarahkoop) [inline AmericanExpress/build.gradle:8]: AGP 8 requires moving the namespace out of `AndroidManifest` into `build.gradle`

File: build.gradle [hunk 2]
```diff
@@ -9,49 +9,34 @@ buildscript {
...
-            "room"           : roomVersion,
+            "kotlin"         : "1.9.10",
+            "androidxTest"   : "1.5.0",
+            "room"           : "2.6.0",
             "playServices"   : "19.2.1",
             "javaSourceCompatibility": sdkTargetJavaVersion,
             "javaTargetCompatibility": sdkTargetJavaVersion,
     ]
-
     ext.deps = [
```
Comment (sarahkoop) [inline build.gradle:20]: I'm not sure if this is the best way to centralize dependencies (Android Studio doesn't recognize this setup, so we don't get the automatic suggestion to upgrade dependencies). In another ticket I'm going to look into alternatives to this.

**Review by sshropshire — APPROVED** [review]
🌊

**Comment by sarahkoop** [comment]
Resolved the AGP 8 Java version issues by running CI with Java 17, but still compiling for Java 11. Open to considering bumping everything to Java 17, but I think 11 is good for now.

**Pull request merged**